### DISCUSSION TITLE
Error testing for "testCantUpdateBoxWithOutGovernance" in my DAO contract

### DISCUSSION THREAD

**Discussion by GenevaRoc** [Cyfrin/foundry-full-course-cu#1221] [General]
Here is the error I'm getting 
`├─ [5037] TimeLock::grantRole(0xb09aa5aeb3702cfd50b6b62bc4532604938f21248a27a1d5ca736082b6819cc1, MyGovernor: [0x5929B14F2984bBE5309c2eC9E7819060C31c970f]) 
    │   └─ ← "AccessControlUnauthorizedAccount(0x6CA6d1e2D5347Bfab1d91e883F1915560e09129D, 0x0000000000000000000000000000000000000000000000000000000000000000)"
    └─ ← "AccessControlUnauthorizedAccount(0x6CA6d1e2D5347Bfab1d91e883F1915560e09129D, 0x0000000000000000000000000000000000000000000000000000000000000000)`

And here is the test code for it 
` bytes32 proposerRole = timeLock.PROPOSER_ROLE();
        bytes32 executorRole = timeLock.EXECUTOR_ROLE();
        bytes32 adminRole = timeLock.DEFAULT_ADMIN_ROLE();

        timeLock.grantRole(proposerRole, address(governor));
        timeLock.grantRole(executorRole, address(0));
        //timeLock.revokeRole(adminRole, USER);
        vm.stopPrank();

        box = new Box(initialOwner);
        box.transferOwnership(address(timeLock));
    }

    function testCantUpdateBoxWithOutGovernance() public {
        vm.expectRevert();
        box.store(1);
    }`

help me out anybody

@theirrationalone

**Comment by FrankezeCode** [comment]
Sup bro, how did you solve this issue ?

**Reply by EngrPips** [reply]
If you are running into this same error you can open a separate discussion and provide every detail that will help us to help you.

**Reply by FrankezeCode** [reply]
Thanks bro, I have resolved mine.

**Reply by EngrPips** [reply]
Interesting. Keep crushing it, Buddy.
